### PR TITLE
WIP: Move testkit inside exonum repository preserving history

### PR DESCRIPTION
**Pull request by DarkEld3r**

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+cache:
+   - '%USERPROFILE%\.cargo'
+
+build_script:
+  - cargo build %cargoflags%
+
+test_script:
+  - cargo test %cargoflags%
+  - cargo run --example timestamping %cargoflags%
+  - cargo run --example configuration_change %cargoflags%
```
Comment (matklad) [inline appveyor.yml]: This'll probably need an `-p test-kit`

File: .travis.yml [hunk 2]
```diff
@@ -89,35 +90,48 @@ install:
...
+  ;;
   "test-core" )
       cargo test --manifest-path exonum/Cargo.toml --verbose --tests --lib --features "float_serialize"
   ;;
   "test-sandbox" )
-      cargo test --manifest-path sandbox/Cargo.toml  --verbose
+      cargo test --manifest-path sandbox/Cargo.toml --verbose
+  ;;
+  "test-testkit" )
+      cargo test --manifest-path testkit/Cargo.toml --verbose
```
Comment (matklad) [inline .travis.yml]: I think it's preferable to use `-p` rather than `--manifest-path` here.

Comment (matklad) [reply .travis.yml]: The same in theory applies to `cargo clippy` and `cargo deadlink`, but I am not sure that those subcommands handle `-p` correctly.

File: .travis.yml [hunk 3]
```diff
@@ -89,35 +90,48 @@ install:
...
   esac
 - cargo install --list
+
 script: |
   case "$FEATURE" in
 
   "fmt" )
       cd exonum && cargo update && cd .. &&
       cd sandbox && cargo update && cd .. &&
+      cd testkit && cargo update && cd .. &&
```
Comment (matklad) [inline .travis.yml]: There's a single `lockfile` per workspace, so just `cargo update && cd sandbox && cargo update` should suffice.

File: .travis.yml [hunk 4]
```diff
@@ -89,35 +90,48 @@ install:
...
   "test-sandbox" )
-      cargo test --manifest-path sandbox/Cargo.toml  --verbose
+      cargo test --manifest-path sandbox/Cargo.toml --verbose
+  ;;
+  "test-testkit" )
+      cargo test --manifest-path testkit/Cargo.toml --verbose
   ;;
   "test-doc" )
-      cargo test --manifest-path sandbox/Cargo.toml  --verbose --doc  -- --test-threads=2
+      cargo test --manifest-path sandbox/Cargo.toml --verbose --doc  -- --test-threads=2
```
Comment (matklad) [inline .travis.yml]: Femtonit: there are two spaces in `--doc  --`.

File: testkit/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+description = """
+Testkit for Exonum blockchain framework, allowing to test service APIs
+synchronously.
+"""
+
+[badges]
+travis-ci = { repository = "exonum/exonum-testkit" }
+
+[dependencies]
+exonum = "0.5.0"
```
Comment (matklad) [inline testkit/Cargo.toml]: This probably should read `exonum = { version = "0.5.0", path = "../exonum" }`

File: testkit/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,33 @@
+[package]
+name = "exonum-testkit"
+version = "0.5.0"
+authors = ["The Exonum Team <exonum@bitfury.com>"]
+homepage = "https://exonum.com/"
+repository = "https://github.com/exonum/exonum-testkit"
```
Comment (matklad) [inline testkit/Cargo.toml]: s/exonum-testkit/exonum/

**Comment by DarkEld3r** [comment]
Closed in favor of #448.

**Pull request closed without merging**